platform-api/src/main/java/io/datacater/core/config/ConfigUtilities.java

```diff
@@ -0,0 +1,118 @@
+package io.datacater.core.config;
+
+import io.datacater.core.authentication.DataCaterSessionFactory;
+import io.datacater.core.config.enums.Kind;
+import io.datacater.core.deployment.DeploymentSpec;
+import io.datacater.core.exceptions.IncorrectConfigException;
+import io.datacater.core.stream.Stream;
+import io.datacater.core.utilities.JsonUtilities;
+import io.smallrye.mutiny.Uni;
+import java.util.*;
+
+public class ConfigUtilities {
+
+  public static Uni<List<ConfigEntity>> getConfig(
+      List<String> configs, DataCaterSessionFactory dsf) {
+    return dsf.withTransaction(
+        (session, transaction) ->
+            session
+                .createQuery("from ConfigEntity", ConfigEntity.class)
+                .getResultList()
+                .onItem()
+                .transform(
+                    x -> x.stream().filter(item -> configs.contains(item.getName())).toList())
+                .onItem()
+                .ifNull()
+                .continueWith(new ArrayList<>()));
+  }
+
+  public static List<String> getConfigNames(Map<String, List<String>> labels) {
+    // TODO consider other label options
+    if (labels != null) {
+      return labels.get("app.datacater.io/name");
+    }
+    return new ArrayList<>();
+  }
+
+  public static Stream combineWithStream(Stream stream, List<ConfigEntity> configList) {
+    if (!configList.isEmpty()) {
```

Comment:
The `if` statement might be obsolete.

Reply:
An Empty list can be given if no config is found and the validation method doesn't "break" if the list is empty, since an empty list would be considered a valid config. So this check is still needed

Reply:
I believe it is not needed because you have only two blocks inside this `if` blocks: The validation and the for loop. The validation does not break because it also checks for an empty list and the for loop would just not execute because the list is empty.

Anyway, this is very very low priority. You can also keep it :)

Reply:
you were right 🙂 i rem oved the if-statement